jgrapht-core/src/main/java/org/jgrapht/alg/scoring/AlphaCentrality.java

```diff
@@ -0,0 +1,240 @@
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.scoring;
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.interfaces.*;
+
+/**
+ * AlphaCentrality implementation.
+ * 
+ * <p>
+ * The <a href="https://en.wikipedia.org/wiki/Alpha_centrality">wikipedia</a> article contains a nice
+ * description of AlphaCentrality. You may also refer to Katz Centrality. See also the following
+ * <a href="https://en.wikipedia.org/wiki/Katz_centrality">page</a>.
+ * </p>
+ * 
+ * <p>
+ * This is a simple iterative implementation of AlphaCentrality which stops after a given number of
+ * iterations or if the AlphaCentrality values between two iterations do not change more than a predefined
+ * value.
+ * </p>
+ *
+ * <p>
+ * Each iteration of the algorithm runs in linear time O(n+m) when n is the number of nodes and m
+ * the number of edges of the graph. The maximum number of iterations can be adjusted by the caller.
+ * The default value is {@link AlphaCentrality#MAX_ITERATIONS_DEFAULT}.
+ * </p>
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Pratik Tibrewal
+ * @since February 2018
+ */
+public final class AlphaCentrality<V, E>
+    implements VertexScoringAlgorithm<V, Double>
+{
+    /**
+     * Default number of maximum iterations.
+     */
+    public static final int MAX_ITERATIONS_DEFAULT = 100;
+
+    /**
+     * Default value for the tolerance. The calculation will stop if the difference of AlphaCentrality
+     * values between iterations change less than this value.
+     */
+    public static final double TOLERANCE_DEFAULT = 0.0001;
+
+    /**
+     * Damping factor default value.
+     */
+    public static final double DAMPING_FACTOR_DEFAULT = 0.01d;
+
+    /**
+     * Exogenous factor default value.
+     */
+    public static final double EXOGENOUS_FACTOR_DEFAULT = 1.0d;
+
+    private final Graph<V, E> g;
+    private Map<V, Double> scores;
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     */
+    public AlphaCentrality(Graph<V, E> g)
+    {
+        this(g, DAMPING_FACTOR_DEFAULT, EXOGENOUS_FACTOR_DEFAULT, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor)
+    {
+        this(g, dampingFactor, EXOGENOUS_FACTOR_DEFAULT, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
```

Comment:
So should I expect that user can enter any one of the above: like either a scalar or a map or I should go for only maps? Instead of vectors I was thinking of using maps as it will be faster. 
If we go for the implementation of both scalar/map, then I think it will be more flexible. If the user enters a scalar I can convert it into a map with all the values as that of the scalar or should I go for writing 2 run functions?

Reply:
Just converting the scalar to map would be fine.

Reply:
Done